aws/eni.go

```diff
@@ -151,14 +151,16 @@ func (s *ENI) attach(ec2Client *ec2.EC2, instanceID string, eniID string) error
 		InstanceId:         aws.String(instanceID),
 		NetworkInterfaceId: aws.String(eniID),
 	}
-	attachment, err := ec2Client.AttachNetworkInterface(attachNetworkInterfaceInput)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-	fmt.Printf("Succefully created attach request. %s\n", attachment.String())
 
 	b := backoff.NewMaxRetries(maxRetries, retryInterval)
 	o := func() error {
+		fmt.Printf("Attempting to attach ENI.\n")
+		attachment, err := ec2Client.AttachNetworkInterface(attachNetworkInterfaceInput)
```

Comment:
Did you test it? I am a little bit afraid that  if the first attempt fails or doesn't succeed immediately  then the second time trying to run attach it will always fail even if its  in state `attaching`

Reply:
I would maybe rather put the whole code into one backoff function 

like:
```
attachENiBackoff:= func() err {

	attachNetworkInterface()
	describeBackoff := func() err {
		describeENI()
		...
	}
 	err := backoff.Retry(describeBackoff)
}

err :=  backoff.Retry(attachENiBackoff)
```

Reply:
no didn't test yet. It's definitely a possibility

Reply:
the AttachNIC  operation is  not atomic, it is a request to attach it and it takes time to execute it

Reply:
I'm testing a debug version of this code that does 2 attach calls in a row see what happens